xc/device.ml

```diff
@@ -1392,6 +1392,81 @@ module Vkbd = struct
 
 end
 
+module Vusb = struct
+  let vusb_controller_plug ~xs ~domid ~driver ~driver_id =
+    let is_running = Qemu.is_running ~xs domid in
+    match is_running with
+    | true  -> (
+      let open Qmp in
+      let qmp_cmd = Command (None, Device_add (driver, driver_id, None)) in
+      qmp_write domid qmp_cmd
+    )
+    | _ ->()
+
+  let vusb_plug ~xs ~domid ~id ~hostbus ~hostport ~version =
+    debug "Vusb plugged: vusb device plugged ";
+    let get_bus v =
+      if String.startswith "1" v then
+         "usb-bus.0"
+      else begin
+          (* Here plug usb controller according to the usb version*)
```

Comment:
What exactly is the version?

Reply:
In fact, it'd be good to know what the range of possible values are for each of the VUSB fields

Reply:
OK, answering my own question here. It's the string representation of the bcdUSB field. See http://www.beyondlogic.org/usbnutshell/usb5.shtml for example. It'll take the values: "1.10", "2.00", "2.50", "3.00", "3.10", and maybe more.

Reply:
yes :)